src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Split.cs

```diff
@@ -66,94 +72,79 @@ public string[] Split(string input, int count, int startat)
         private static string[] Split(Regex regex, string input, int count, int startat)
         {
             if (count < 0)
-                throw new ArgumentOutOfRangeException(nameof(count), SR.CountTooSmall);
-            if (startat < 0 || startat > input.Length)
-                throw new ArgumentOutOfRangeException(nameof(startat), SR.BeginIndexNotNegative);
-
-            string[] result;
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.count, ExceptionResource.CountTooSmall);
+            }
+            if ((uint)startat > (uint)input.Length)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.startat, ExceptionResource.BeginIndexNotNegative);
+            }
 
             if (count == 1)
             {
-                result = new string[1];
-                result[0] = input;
-                return result;
+                return new[] { input };
             }
 
-            count -= 1;
+            count--;
+            var state = (al: new List<string>(), prevat: 0, input, count);
```

Comment:
I know that the original variable was named the same, but what does `al` stand for? The only thing I can guess is "a list".

Reply:
Heh, good question, no idea.  I'll change it.